types/address.go

```diff
@@ -674,12 +676,24 @@ func GetPubKeyFromBech32(pkt Bech32PubKeyType, pubkeyStr string) (crypto.PubKey,
 		return nil, err
 	}
 
-	pk, err := cryptocodec.PubKeyFromBytes(bz)
+	aminoPk, err := cryptocodec.PubKeyFromBytes(bz)
 	if err != nil {
 		return nil, err
 	}
 
-	return pk, nil
+	var protoPk crypto.PubKey
+	switch aminoPk.(type) {
+	case tmed25519.PubKey:
+		protoPk = &ed25519.PubKey{
+			Key: aminoPk.Bytes(),
+		}
+
+	default:
+		// We only allow ed25519 pubkeys to be bech32-ed right now.
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidType, "bech32 expects %T, got %T", tmed25519.PubKey{}, aminoPk)
+	}
+
+	return protoPk, nil
```

Comment:
From Marie:
> the remaining issues seem primarily to be due to the fact that some tmcrypto.PubKey (returned by GetPubKeyFromBech32) is provided to MsgCreateValidator and since it does not implement proto.Message , this fails here https://github.com/cosmos/cosmos-sdk/blob/11924ddb16450906e44418a62005107ba9ddfedb/x/staking/types/msg.go#L38

Trying to fix that here.